app/src/main/java/com/example/drawingactivity/MainActivity.kt

```diff
@@ -17,11 +25,83 @@ class MainActivity : AppCompatActivity() {
         viewModel.currentBitmap?.let { drawableArea.setBitmap(it) }
         val colorSelector = findViewById<PrimaryColorCircleView>(R.id.colorCircle)
         colorSelector.onChangeColor = { viewModel.cyclePrimaryColor() }
+        findViewById<Button>(R.id.add_layer).setOnClickListener {
+            colorPickerDialog()
+        }
+
+        if (viewModel.isDialogOpen) {
+            colorPickerDialog()
+        }
+
+        viewModel.onAddLayerSuccess = {
+            Toast.makeText(this, getString(R.string.addLayerSuccess) + it, Toast.LENGTH_SHORT)
+                .show()
+        }
 
         viewModel.onUpdate = {
             colorSelector.circlePrimaryColor = it.primaryColor
             drawableArea.pathPaint.color = it.primaryColor
             drawableArea.userDrawnBitmap = it.currentBitmap
         }
     }
+
+    @SuppressLint("ClickableViewAccessibility")
+    private fun colorPickerDialog() {
+        viewModel.isDialogOpen = true
+        val inflater = LayoutInflater.from(this@MainActivity)
+        val view = inflater.inflate(R.layout.add_layer_dialog, null)
+        val dialog = AlertDialog.Builder(this).apply {
+            setTitle(context.getString(R.string.addLayerDialogTitle))
+            setView(view)
+            val errorView = view.findViewById<TextView>(R.id.errorText)
+            val layerNameInput = view.findViewById<EditText>(R.id.layerNameField)
+            layerNameInput.hint =
+                context.getString(R.string.addLayerDefaultName, viewModel.layerNumber)
+            if (viewModel.isDialogOpen && viewModel.currentUserEnteredLayerName != null) {
+                layerNameInput.setText(viewModel.currentUserEnteredLayerName)
+                if (layerNameInput.text.toString().length > NAME_LENGTH) {
```

Comment:
You can reduce this.
`errorView.isVisible = layerNameInput.text.toString().length > NAME_LENGTH`

Reply:
reduced to given statement